torch/_dynamo/compiled_autograd.py

```diff
@@ -490,22 +490,49 @@ def set_node_origin(
 in_compiled_autograd_region = False
 
 
-@contextlib.contextmanager
-def enable(compiler_fn):
-    prior = torch._C._dynamo.compiled_autograd.set_autograd_compiler(
-        functools.partial(AutogradCompilerInstance, compiler_fn)
-    )
-    if snapshot_verbose_logging_enabled():
-        torch._C._dynamo.compiled_autograd.set_verbose_logger(cpp_verbose_log_fn)
-    global compiled_autograd_enabled
-    compiled_autograd_enabled = True
-    try:
-        with torch.autograd.set_multithreading_enabled(False):
-            yield
-    finally:
-        if not prior:
-            compiled_autograd_enabled = False
-        torch._C._dynamo.compiled_autograd.set_autograd_compiler(prior)
+class _EnableContext:
+    def __init__(self, compiler_fn):
+        self.compiler_fn = compiler_fn
+        self.warmup_count = 0
```

Comment:
this limits the use of this context manager, in order to support this stateful class, users must reuse the same context manager when they want the warmup count to be respected

in cases where they should have different warmup counts, the user needs to create a new context manager

I haven't thought of an alternative yet, but if there's none, we should at least add a warning on the usage

Reply:
I added a comment above the `torch._dynamo.config.warmup_runs` config - when users go in to change that config, they should be able to see the comment about persisting warmup count for compiled autograd.

(I also thought about always printing a warning in compiled autograd whenever `config.warmup_count > 0`, but I think users also have a habit of ignoring warnings, so putting a comment above the config might be sufficient/better).